app/services/fast_track/hypertension_special_issue_manager.rb

```diff
@@ -8,27 +8,31 @@ def initialize(submission)
       @submission = submission
     end
 
-    def add_special_issue
-      data = JSON.parse(submission.form_json, symbolize_names: true)
-      disabilities = data[:form526][:form526][:disabilities]
-      data[:form526][:form526][:disabilities] = add_rrd_to_disabilities(disabilities)
-      submission.update!(form_json: JSON.dump(data))
+    def submission_data
+      @submission_data ||= JSON.parse(submission.form_json, symbolize_names: true)
     end
 
-    def add_rrd_to_disabilities(disabilities)
-      disabilities.each do |da|
-        add_rrd(da) if da[:diagnosticCode] == 7101 && da[:disabilityActionType].downcase == 'increase'
-      end
-      disabilities
+    def disabilities
+      @disabilities ||= submission_data[:form526][:form526][:disabilities]
     end
 
-    def add_rrd(disability)
-      rrd_code = 'RRD'
-      if disability[:specialIssues].blank?
-        disability[:specialIssues] = [rrd_code]
-      elsif !disability[:specialIssues].include? rrd_code
-        disability[:specialIssues].append(rrd_code)
+    def add_special_issue
+      disabilities.each do |disability|
+        add_rrd_code(disability) if hypertension_increase?(disability)
       end
+      submission.update!(form_json: JSON.dump(submission_data))
```

Comment:
`disabilities` and `submission_data` are now lazily-initialized, private attributes.